### PR TITLE
Fixes #95 - array returned from intrinsic.

### LINKED ISSUE #95: parts.join is not a function when getting Custom Resource outputs
Ab abstract from CFN template from http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-lambda.html.
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  AllSecurityGroups:
    Type: Custom::Split
    Properties:
      ServiceToken: "somefunctionarn"

Outputs:
  AllSecurityGroups:
    Description: Security Groups that are associated with the EC2 instance
    Value:
      Fn::Join:
      - ", "
      - Fn::GetAtt:
        - AllSecurityGroups
        - Value
```

Results in:
```
2017-11-19T18:03:19.296Z - error: uncaughtException: parts.join is not a function date=Sun Nov 19 2017 18:03:19 GMT+0000 (UTC), pid=122, uid=0, gid=0, cwd=/, execPath=/usr/local/bin/node, version=v9.2.0, argv=[/usr/local/bin/node, /usr/local/bin/cfn-lint, validate, out3.yaml], rss=43716608, heapTotal=22614016, heapUsed=14802432, external=123909, loadavg=[1.1455078125, 1.27294921875, 1.2451171875], uptime=3872, trace=[column=18, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=fnJoin, line=926, method=null, native=false, column=16, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=doIntrinsicJoin, line=557, method=null, native=false, column=20, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=resolveIntrinsicFunction, line=482, method=null, native=false, column=38, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=recursiveDecent, line=434, method=null, native=false, column=13, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=recursiveDecent, line=444, method=null, native=false, column=13, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=recursiveDecent, line=444, method=null, native=false, column=13, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=recursiveDecent, line=444, method=null, native=false, column=5, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=resolveReferences, line=404, method=null, native=false, column=5, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=validateWorkingInput, line=144, method=null, native=false, column=11, file=/usr/local/lib/node_modules/cfn-lint/src/validator.ts, function=Object.validateFile, line=63, method=validateFile, native=false], stack=[TypeError: parts.join is not a function,     at fnJoin (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:926:18),     at doIntrinsicJoin (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:557:16),     at resolveIntrinsicFunction (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:482:20),     at recursiveDecent (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:434:38),     at recursiveDecent (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:444:13),     at recursiveDecent (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:444:13),     at recursiveDecent (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:444:13),     at resolveReferences (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:404:5),     at validateWorkingInput (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:144:5),     at Object.validateFile (/usr/local/lib/node_modules/cfn-lint/src/validator.ts:63:11)]
```
cfn-lint@v1.3.4
https://github.com/martysweet/cfn-lint/blob/fc3139c69d39853dfd8d9feec2e07c995b19bb30/src/validator.ts#L926

@akdor1154 I'm guessing this is due to the custom resource mocking? The below seems to be valid.
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  AllSecurityGroups:
    Type: Custom::Split
    Properties:
      ServiceToken: "somefunctionarn"

Outputs:
  AllSecurityGroups:
    Description: Security Groups that are associated with the EC2 instance
    Value:
      Fn::Join:
        - ","
        - - "a"
          - "b"
```

### PR DESCRIPTION
**Pull request by RazzM13**

### CODE REVIEW

File: src/validator.ts [hunk 1]
```diff
@@ -592,6 +592,12 @@ function doIntrinsicJoin(ref: any, key: string){
         // Specify this as an invalid string
         return "INVALID_JOIN";
     }else{
+        // Instrinsic functions that return an array of values
+        // must be patched to be resolved
+        if ((RegExp(/^Fn::/gi).test(key)) && (!Array.isArray(parts))) {
```
Comment (martysweet) [inline src/validator.ts]: Can we use .startsWith() here?

Comment (RazzM13) [reply src/validator.ts]: Sure.

**Comment by RazzM13** [comment]
Please have a look at my last comment from #95 as to why I believe that this is no longer applicable.

**Comment by RazzM13** [comment]
I'm closing this PR as #135 supersedes it.

**Pull request closed without merging**